ksql-test-commons/src/test/java/io/confluent/ksql/test/commons/TestCaseTest.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.commons;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.metastore.model.KsqlTopic;
+import io.confluent.ksql.serde.json.KsqlJsonTopicSerDe;
+import io.confluent.ksql.util.KafkaTopicClient;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.streams.TopologyTestDriver;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestCaseTest {
+
+  @Rule
+  public final org.junit.rules.ExpectedException expectedException = org.junit.rules.ExpectedException
+      .none();
+
+  private final SerdeSupplier serdeSupplier = new StringSerdeSupplier();
+  private final Topic topic = new Topic("foo_kafka", Optional.empty(), serdeSupplier, 4, 1);
+  private final Record record = new Record(topic, "k1", "v1, v2", 123456789L, null);
+  private final TestCase testCase = new TestCase(
+      null,
+      "test",
+      Collections.emptyMap(),
+      ImmutableList.of(topic),
+      ImmutableList.of(record),
+      ImmutableList.of(record),
+      Collections.emptyList(),
+      ExpectedException.none(),
+      PostConditions.NONE
+  );
+
+
+  @Mock
+  private TopologyTestDriver topologyTestDriver;
+
+
+  @Before
+  public void init() {
+
+  }
+
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldProcessInputRecords() {
+    // Given:
+    final TopologyTestDriverContainer topologyTestDriverContainer = getSampleTopologyTestDriverContainer();
+
+
+    // When:
+    testCase.processInput(topologyTestDriverContainer, null);
+
+
+    // Then:
+    verify(topologyTestDriver).pipeInput(any(ConsumerRecord.class));
```

Comment:
maybe verify the exact record we expect?

Reply:
Good point, versifying the record too.